integration/spark/app/src/main/java/io/openlineage/spark/agent/OpenLineageSparkListener.java

```diff
@@ -104,6 +138,33 @@ private static void sparkSQLExecEnd(SparkListenerSQLExecutionEnd endEvent) {
   /** called by the SparkListener when a job starts */
   @Override
   public void onJobStart(SparkListenerJobStart jobStart) {
+    Optional<Long> eventProcessingTimeoutSeconds =
+        OpenLineageSparkListener.getEventProcessingTimeoutSeconds();
+
+    if (!eventProcessingTimeoutSeconds.isPresent()) {
```

Comment:
Can we move this to wrapper method rather than calling it in every event processing method? As in, modify `callWithTimeout` to do this check